### PR TITLE
feat(pages.test.js): add tests to detect changes

### PR DESCRIPTION
**Pull request by MarquiseRosier**
detect changes in helix pages

I do not expect this to build; just wanna make sure we can have conversations about some tech choices :D

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -247,6 +247,30 @@ jobs:
...
+  run_diff_tests:
+    executor: node10
+    steps:
+      - setup
+
+      - run: mkdir junit
+
+      - run:
+        name: Load environment variables
+        command: cat env/TEST_DOMAIN >> $BASH_ENV
```
Comment (trieloff) [inline .circleci/config.yml:260]:
```suggestion
          name: Load environment variables
          command: cat env/TEST_DOMAIN >> $BASH_ENV
```

I think you need to indent this.

File: package.json [hunk 2]
```diff
@@ -33,6 +34,7 @@
   },
   "devDependencies": {
     "@adobe/eslint-config-helix": "1.1.1",
+    "@adobe/helix-fetch": "^1.6.1",
```
Comment (rofe) [inline package.json:37]:
```suggestion
    "@adobe/helix-fetch": "1.6.1",
```
Use pinned dependencies

File: package.json [hunk 3]
```diff
@@ -44,6 +46,7 @@
     "eslint-plugin-header": "3.0.0",
     "eslint-plugin-import": "2.21.1",
     "fs-extra": "9.0.1",
+    "html-differ": "^1.4.0",
```
Comment (rofe) [inline package.json:49]:
```suggestion
    "html-differ": "1.4.0",
```
Use pinned dependencies

Comment (trieloff) [inline package.json:49]: Better use the diff tool in helix shared

Comment (MarquiseRosier) [reply package.json:49]: hey @rofe I think you need to replace the test:pages in package.json; step where TEST_DOMAIN=$TEST_DOMAIN to hlx.page! This is done in the other tests; or set the process.env.TEST_DOMAIN!

Comment (rofe) [reply package.json:49]: Thanks for the hint, the test is passing now!

File: test/pages/pages.test.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  const json = {
+    limit: 2,
+    threshold: 400,
+  };
+  const method = 'post';
+  const res = await fetch('https://adobeioruntime.net/api/v1/web/helix/helix-services/run-query@v2/most-visited', { method, json });
+  if (!res.ok) {
+    assert.fail('test setup failed to gather test urls');
+  }
+  base_urls = (await res.json()).results;
```
Comment (trieloff) [inline test/pages/pages.test.js:57]: I think you should always read the body of the response, so I'd pull this line above the `assert.fail`

File: test/pages/pages.test.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  // construct array of promises from fetch
+  changes = base_urls.map((obj) => {
+    // eslint-disable-next-line camelcase
+    const { req_url } = obj;
+    const { pathname, hostname } = new URL(req_url);
+    const headers = {
```
Comment (tripodsan) [inline test/pages/pages.test.js:41]: this is no longer needed with helix-fetch v1.6.2: https://github.com/adobe/helix-fetch/releases/tag/v1.6.2

**Comment by MarquiseRosier** [comment]
https://github.com/adobe/theblog-tests/pull/16 needs to be merged for this

**Comment by tripodsan** [comment]
@MarquiseRosier the new tests seem to timeout:
https://circleci.com/gh/adobe/helix-pages/4748?utm_campaign=workflow-failed&utm_medium=email&utm_source=notification

can you increase the timeout ?

**Pull request merged**